src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs

```diff
@@ -313,5 +321,159 @@ public static bool TryParse(string value, out SemanticVersion result)
         {
             return !(b < a);
         }
+
+        // private methods
+        private bool IsServerVersion(string preRelease)
```

Comment:
Rename to `IsServerInternalBuildVersion` (or `IsServerInternalBuild`)

Reply:
`IsServerVersion` is removed and the `ServerVersion` detection changed to `TryParse()` instead, which relaxes the conversion and allows to accept existing `SemanticVersion` instances like `new SemanticVersion(4, 0, 0, "")`.